### PR TITLE
[CPDLP-2470] Change to / update of SIT does not touch updated_at field of partnership (or school response)

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

- Ticket: https://dfedigital.atlassian.net/browse/CPDLP-2470

### Changes proposed in this pull request

* Updated `Api::V3::ECF::PartnershipSerializer`
* Updated `Api::V3::ECF::PartnershipsQuery`
* Updated release notes

### CODE REVIEW

File: app/serializers/api/v3/ecf/partnership_serializer.rb [hunk 1]
```diff
@@ -51,7 +51,12 @@ class PartnershipSerializer
         end
 
         attribute :updated_at do |partnership|
-          partnership.updated_at.rfc3339
+          [
+            partnership.updated_at,
+            partnership.school.updated_at,
+            partnership.delivery_partner&.updated_at,
+            partnership.school&.induction_coordinator_profiles&.first&.updated_at,
```
Comment (cwrw) [inline app/serializers/api/v3/ecf/partnership_serializer.rb:58]: could we use `partnership.school&.induction_coordinators&.first&.updated_at` instead as that's what's being updated (the user)?

File: app/services/api/v3/ecf/partnerships_query.rb [hunk 2]
```diff
@@ -20,7 +20,15 @@ def partnerships
             .where(partnerships: { cohort: cohorts })
             .order(sort_order(default: "partnerships.created_at ASC", model: Partnership))
             .distinct
-          scope = scope.where("partnerships.updated_at > ?", updated_since) if updated_since_filter.present?
+
+          if updated_since_filter.present?
+            scope = scope.includes(:delivery_partner, school: [:induction_coordinator_profiles])
```
Comment (cwrw) [inline app/services/api/v3/ecf/partnerships_query.rb:25]:
```suggestion
            scope = scope.includes(:delivery_partner, school: [:induction_coordinators])
```

File: app/services/api/v3/ecf/partnerships_query.rb [hunk 3]
```diff
@@ -20,7 +20,15 @@ def partnerships
...
+            scope = scope.where(partnerships: { updated_at: updated_since.. })
+              .or(scope.where(school: { updated_at: updated_since.. }))
+              .or(scope.where(delivery_partner: { updated_at: updated_since.. }))
+              .or(scope.where(school: { induction_coordinator_profiles: { updated_at: updated_since.. } }))
```
Comment (cwrw) [inline app/services/api/v3/ecf/partnerships_query.rb:29]:
```suggestion
              .or(scope.where(school: { users: { updated_at: updated_since.. }}))
```
Induction coordinator names/emails (users) are what are changing so we have to surface those values :)

File: app/serializers/api/v3/ecf/partnership_serializer.rb (hunk 1 above)
Comment (mooktakim) [reply app/serializers/api/v3/ecf/partnership_serializer.rb]: fixed 👍

**Review by leandroalemao — APPROVED** [review]
Tks mook .. LGTM 👍 😄

**Pull request merged**